CONTRIBUTING.md

```diff
@@ -177,6 +177,32 @@ In the case of a pseudo-local variable (a field used briefly to pass around a lo
 injections of said method), the field should be named with the namespace first, then the name of the injected method,
 and finally the name of the local (`quilt$injectedMethod$localName`).
 
+## Gradle Conventions
+
+### Declaring dependencies between modules
+
+In the `qslModule` extension, there is a `moduleDependencies` field. Dependencies are declared in a tree like structure reflecting how QSL libraries and modules are layed out.
+This field can be configured like:
+```groovy
+qslModule {
+    // ...
+    moduleDependencies {
+        library_name { // The QSL Library to depend on
+            module_name { // The QSL Module in `library_name` to depend on
+                api() // Default behavior, should not need to be called.
+            }
+            module_2 {
+                testmod() // Only depend on this module when testing the code
+            }
+            module_3 {
+                impl() // This library is only used to compile this module and is not part of the API. This should be rarely used.
```

Comment:
i don't think that statement is true, it would be better to explain when to use it (which is probably pretty often)
```suggestion
                // By default, dependencies are `api` dependencies; they are put on the classpath
                // of mods that depend on this module.
                // Impl dependencies are dependencies that are only used internally in a module,
                // and classes from it are never exposed through this module's public API.
                // For example, a dependency on lifecyle events would usually be impl
                impl() 
```